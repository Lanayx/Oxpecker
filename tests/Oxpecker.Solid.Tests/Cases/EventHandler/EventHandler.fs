module Oxpecker.Solid.Tests.Cases.EventHandler

open Browser.Types
open Oxpecker.Solid
open Browser

let clicked2 = fun (evt: MouseEvent) -> console.log(evt.``type``)

[<SolidComponent>]
let Test () =
    div(onClick = fun _ -> console.log("clicked1")) {
        h1(onClick = clicked2) {
            "Hello"
        }
        h2()
            .on("click", fun _ -> console.log("clicked3"))
            .on("change", fun _ -> console.log("changed"))
            .on("load", fun _ -> console.log("loaded")){
            "Hello"
        }
        h3(id="test", onClick = fun _ -> console.log("clicked4"))
            .on("change", fun _ -> console.log("changed"))
            .on("load", fun _ -> console.log("loaded"))
    }
