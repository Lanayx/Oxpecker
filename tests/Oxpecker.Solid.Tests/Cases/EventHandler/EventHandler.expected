import { some } from "./fable_modules/fable-library-js.4.21.0/Option.js";

export function clicked2(evt) {
    console.log(some(evt.type));
}

export function Test() {
    return <div onClick={(_arg_6) => {
            console.log(some("clicked1"));
        }}>
        <h1 onClick={(evt) => {
                clicked2(evt);
            }}>
            Hello
        </h1>
        <h2 on:click={(_arg) => {
                console.log(some("clicked3"));
            }}
            on:change={(_arg_1) => {
                console.log(some("changed"));
            }}
            on:load={(_arg_2) => {
                console.log(some("loaded"));
            }}>
            Hello
        </h2>
        <h3 id="test"
            onClick={(_arg_3) => {
                console.log(some("clicked4"));
            }}
            on:change={(_arg_4) => {
                console.log(some("changed"));
            }}
            on:load={(_arg_5) => {
                console.log(some("loaded"));
            }}></h3>
    </div>;
}

